decoder/vaapidecoder_h264.cpp

```diff
@@ -1637,6 +1647,19 @@ YamiStatus VaapiDecoderH264::ensureContext(const SharedPtr<SPS>& sps)
             return status;
         // return YAMI_DECODE_FORMAT_CHANGE to info upper layer va context changed
         return YAMI_DECODE_FORMAT_CHANGE;
+    }else if(isResolutionChanged(sps)){
+        m_configBuffer.width = sps->frame_cropping_flag ? sps->m_cropRectWidth
```

Comment:
we can change it to:
if (isResolutionChanged(sps)){
         m_configBuffer.width = sps->frame_cropping_flag ? sps->m_cropRectWidth
                                                         : sps->m_width;
         m_configBuffer.height = sps->frame_cropping_flag ? sps->m_cropRectHeight
                                                          : sps->m_height;
         m_configBuffer.surfaceWidth = sps->m_width;
         m_configBuffer.surfaceHeight = sps->m_height;
         m_configBuffer.flag |= HAS_SURFACE_NUMBER;
         m_configBuffer.profile
             = VAProfileH264High; // FIXME: set different profile later
if (if (isDecodeContextChanged(sps))){
   status = VaapiDecoderBase::terminateVA();
status = VaapiDecoderBase::start(&m_configBuffer);
         if (status != YAMI_SUCCESS)
              return status;
}
 return YAMI_DECODE_FORMAT_CHANGE;
}